File: dailyQuestion/2020-05/05-19/python/solution.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) - 1
        count = 1
        flag = True
        while i <= j:
            if s[i] == s[j]:
                i += 1
                j -= 1
            else:
                flag = self.validate(s[i:j]) or self.validate(s[i + 1:j + 1])
                break
        return flag

    def validate(self, s: str) -> bool:
        i = 0
        j = len(s) - 1
        while i <= j:
            if s[i] != s[j]:
                return False
            i += 1
            j -= 1
        return True

if __name__ == "__main__" :
    s = "ebcbbececabbacecbbcbe"

    a = "abcdef"
    c = a[1:2]
    result = validPalindrome(s)
    print(result)

File: dailyQuestion/2020-05/05-19/golang/solution.go
package main

import (
	"fmt"
)

func main() {
	a := "ebcbbececabbacecbbcbe"
	b := validPalindrome(a)
	fmt.Println(b)
}

func validPalindrome(s string) bool {
	i, j := 0, len(s)-1
	flag := true
	for i <= j {
		if s[i] == s[j] {
			i += 1
			j -= 1
		} else {
			flag = validate(s[i:j]) || validate(s[i+1:j+1])
			break
		}
	}
	return flag
}

func validate(s string) bool {
	i, j := 0, len(s)-1
	for i <= j {
		if s[i] != s[j] {
			return false
		}
		i += 1
		j -= 1
	}
	return true
}
